src/components/Chart/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useMemo, useState } from "react";
+import { DetailsTooltip } from "@stellar/design-system";
+import { VictoryArea, VictoryAxis, VictoryChart, VictoryGroup } from "victory";
+import { formatAmount } from "helpers/formatAmount";
+import { ChartData } from "types/types.d";
+
+import "./styles.scss";
+
+type TimeFrames = { label: string; segments: number | undefined }[];
+
+interface ChartProps {
+  chartData: ChartData[];
+  colorRgb: number[];
+  header: {
+    title: string;
+    description: string;
+    h1Color?: string;
+  };
+  timeframes?: TimeFrames;
+}
+
+export const Chart = ({
+  chartData,
+  colorRgb,
+  header,
+  timeframes = [],
+}: ChartProps) => {
+  const customStyle = {
+    ...(header.h1Color ? { "--Chart-header-color": header.h1Color } : {}),
+  } as React.CSSProperties;
+  const allTimeframes = useMemo(
+    () => [...timeframes, { label: "ALL", segments: undefined }],
+    [timeframes],
+  );
+  const rgbStr = colorRgb.toString();
+
+  const isTimeframeSelectorRequired = allTimeframes.length > 1;
+
+  const [selectedData, setSelectedData] = useState(chartData);
+  const [currentTimeframe, setCurrentTimeframe] = useState(
+    allTimeframes.length - 1,
+  );
+
+  useEffect(() => {
+    if (isTimeframeSelectorRequired) {
+      setSelectedData(
+        chartData.slice(0, allTimeframes[currentTimeframe].segments),
+      );
+    }
+  }, [allTimeframes, chartData, currentTimeframe, isTimeframeSelectorRequired]);
+
+  const generateTimeframeButtons = () =>
+    allTimeframes.map((t, i) => (
+      <button
+        className={`Chart__header__button--${
+          currentTimeframe === i ? "active" : ""
+        } Chart__header__button`}
+        key={t.label}
+        onClick={() => setCurrentTimeframe(i)}
+      >
+        {t.label}
+      </button>
+    ));
+
+  if (!chartData.length) {
+    return null;
+  }
+
+  const chartNote = chartData[0].y - chartData[1].y;
+  const isChartNoteNegative = chartNote < 0;
+  const chartNotePercentage = (
+    (Math.abs(chartNote) / chartData[1].y) *
+    100
+  ).toFixed(2);
+
+  return (
+    <section className="Chart" style={customStyle}>
+      <header className="Chart__header">
+        <div className="Chart__header__row">
+          <div className="Chart__header__label">
+            <DetailsTooltip
```

Comment:
We can use the updated `Tooltip` component here ([example](https://github.com/stellar/amm-reference-ui/blob/main/src/components/StatItem/index.tsx#L26:L32)).